jenkins/Jenkinsfile

```diff
@@ -0,0 +1,477 @@
+#!/bin/groovy
+
+// global variables
+repository = "xamarin/xamarin-macios"
+isPr = false
+branchName = null
+gitHash = null
+packagePrefix = null
+virtualPath = null
+xiPackageUrl = null
+xmPackageUrl = null
+utils = null
+errorMessage = null
+currentStage = null
+
+xiPackageFilename = null
+xmPackageFilename = null
+manifestFilename = null
+reportPrefix = null
+createFinalStatus = true
+skipLocalTestRunReason = ""
+
+def abortExecutingBuilds ()
+{
+    // This runs into problems with the Jenkins sandbox:
+    //      org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method jenkins.model.Jenkins getItemByFullName java.lang.String
+    // so disable for now.
+
+    // def job = Jenkins.instance.getItemByFullName (env.JOB_NAME)
+    // for (build in job.builds) {
+    //     if (!build.isBuilding ())
+    //         continue
+
+    //     if (build.number > currentBuild.number) {
+    //         error ("There is already a newer build in progress (#${build.number})")
+    //     } else if (build.number < currentBuild.number) {
+    //         def exec = build.getExecutor ()
+    //         if (exec == null) {
+    //             echo ("No executor for build ${build.number}")
+    //         } else {
+    //             exec.interrupt (Result.ABORTED, new CauseOfInterruption.UserInterruption ("Aborted by build #${currentBuild.number}"))
+    //             echo ("Aborted previous build: #${build.number}")
+    //         }
+    //     }
+    // }
+}
+
+github_pull_request_info = null
+def githubGetPullRequestInfo ()
+{
+    if (github_pull_request_info == null && isPr) {
+        withCredentials ([string (credentialsId: 'macios_github_comment_token', variable: 'GITHUB_PAT_TOKEN')]) {
+            def url = "https://api.github.com/repos/${repository}/pulls/${env.CHANGE_ID}"
+            def outputFile = ".github-pull-request-info.json"
+            try {
+                sh ("curl -vf -H 'Authorization: token ${GITHUB_PAT_TOKEN}' --output '${outputFile}' '${url}'")
+                github_pull_request_info = readJSON (file: outputFile)
+                echo ("Got pull request info: ${github_pull_request_info}")
```

Comment:
Do we want to log the exception for easier debugging?

Reply:
The exception is not caught, it will bubble up and eventually fail the job.